lib/websocket.js

```diff
@@ -392,8 +392,13 @@ class WebSocket extends EventEmitter {
   }
 }
 
+WebSocket.CONNECTING = 0
+WebSocket.OPEN = 1
+WebSocket.CLOSING = 2
+WebSocket.CLOSED = 3
+
 readyStates.forEach((readyState, i) => {
-  const descriptor = { enumerable: true, value: i };
+  const descriptor = { enumerable: true, value: i, configurable: false, writable: false };
```

Comment:
Why this change? `false` is the default value for `configurable` and `writable`.

Reply:
when i assigned the `WebSocket.CLOSING = 2` it probably changed the configurable to true, so i had to override it again - test was failing if i didn't do this

Reply:
What test? Behavior is tested here https://github.com/websockets/ws/blob/0ad1f9d6a48ed1b30bda09b958cb142c1e09cced/test/websocket.test.js#L102-L111.

Reply:
here is my test log if i removed conf + writable

```diff
  Constants
      `CONNECTING`
        1) is enumerable property of class
        ✓ is enumerable property of prototype
      `OPEN`
        2) is enumerable property of class
        ✓ is enumerable property of prototype
      `CLOSING`
        3) is enumerable property of class
        ✓ is enumerable property of prototype
      `CLOSED`
        4) is enumerable property of class
        ✓ is enumerable property of prototype
    Attributes
      `binaryType`
        ✓ is enumerable and configurable
        ✓ defaults to 'nodebuffer'
        ✓ can be changed to 'arraybuffer' or 'fragments'
      `bufferedAmount`
        ✓ is enumerable and configurable
        ✓ defaults to zero
        ✓ defaults to zero upon "open"
        ✓ takes into account the data in the sender queue
        ✓ takes into account the data in the socket queue
      `extensions`
        ✓ is enumerable and configurable
        ✓ exposes the negotiated extensions names (1/2)
        ✓ exposes the negotiated extensions names (2/2)
      `protocol`
        ✓ is enumerable and configurable
        ✓ exposes the subprotocol selected by the server
      `readyState`
        ✓ is enumerable and configurable
        ✓ defaults to `CONNECTING`
        ✓ is set to `OPEN` once connection is established
        ✓ is set to `CLOSED` once connection is closed
        ✓ is set to `CLOSED` once connection is terminated
      `url`
        ✓ is enumerable and configurable
        ✓ exposes the server url
    Events
      ✓ emits an 'error' event if an error occurs
      ✓ does not re-emit `net.Socket` errors
      ✓ emits an 'upgrade' event
      ✓ emits a 'ping' event
      ✓ emits a 'pong' event
    Connection establishing
      ✓ fails if the Sec-WebSocket-Accept header is invalid
      ✓ close event is raised when server closes connection
      ✓ error is emitted if server aborts connection
      ✓ unexpected response can be read when sent by server
      ✓ request can be aborted when unexpected response is sent by server
      ✓ fails if the opening handshake timeout expires (102ms)
      ✓ fails if an unexpected Sec-WebSocket-Extensions header is received
      ✓ fails if the Sec-WebSocket-Extensions header is invalid (1/2)
      ✓ fails if the Sec-WebSocket-Extensions header is invalid (2/2)
      ✓ fails if an unexpected extension is received (1/2)
      ✓ fails if an unexpected extension is received (2/2)
      ✓ fails if server sends a subprotocol when none was requested
      ✓ fails if server sends an invalid subprotocol
      ✓ fails if server sends no subprotocol
      ✓ does not follow redirects by default
      ✓ honors the `followRedirects` option
      ✓ honors the `maxRedirects` option
    Connection with query string
      ✓ connects when pathname is not null
      ✓ connects when pathname is null
    #ping
      ✓ throws an error if `readyState` is `CONNECTING`
      ✓ increases `bufferedAmount` if `readyState` is 2 or 3
      ✓ calls the callback w/ an error if `readyState` is 2 or 3
      ✓ can send a ping with no data
      ✓ can send a ping with data
      ✓ can send numbers as ping payload
      ✓ throws an error if the data size is greater than 125 bytes
    #pong
      ✓ throws an error if `readyState` is `CONNECTING`
      ✓ increases `bufferedAmount` if `readyState` is 2 or 3
      ✓ calls the callback w/ an error if `readyState` is 2 or 3
      ✓ can send a pong with no data
      ✓ can send a pong with data
      ✓ can send numbers as pong payload
      ✓ throws an error if the data size is greater than 125 bytes
    #send
      ✓ throws an error if `readyState` is `CONNECTING`
      ✓ increases `bufferedAmount` if `readyState` is 2 or 3
      ✓ calls the callback w/ an error if `readyState` is 2 or 3
      ✓ can send a big binary message (145ms)
      ✓ can send text data
      ✓ does not override the `fin` option
      ✓ sends numbers as strings
      ✓ can send binary data as an array
      ✓ can send binary data as a buffer
      ✓ can send an `ArrayBuffer`
      ✓ can send a `Buffer`
      ✓ calls the callback when data is written out
      ✓ works when the `data` argument is falsy
      ✓ honors the `mask` option
    #close
      ✓ closes the connection if called while connecting (1/3)
      ✓ closes the connection if called while connecting (2/3) (306ms)
      ✓ closes the connection if called while connecting (3/3)
      ✓ can be called from an error listener while connecting
      ✓ can be called from a listener of the 'upgrade' event
      ✓ throws an error if the first argument is invalid (1/2)
      ✓ throws an error if the first argument is invalid (2/2)
      ✓ throws an error if the message is greater than 123 bytes
      ✓ sends the close status code only when necessary
      ✓ works when close reason is not specified
      ✓ works when close reason is specified
      ✓ permits all buffered data to be delivered
      ✓ allows close code 1013
      ✓ allows close code 1014
      ✓ does nothing if `readyState` is `CLOSED`
      ✓ sets a timer for the closing handshake to complete
    #terminate
      ✓ closes the connection if called while connecting (1/2)
      ✓ closes the connection if called while connecting (2/2) (304ms)
      ✓ can be called from an error listener while connecting
      ✓ can be called from a listener of the 'upgrade' event
      ✓ does nothing if `readyState` is `CLOSED`
    WHATWG API emulation
      ✓ supports the `on{close,error,message,open}` attributes
      ✓ works like the `EventEmitter` interface
      ✓ doesn't return listeners added with `on`
      ✓ doesn't remove listeners added with `on`
      ✓ adds listeners for custom events with `addEventListener`
      ✓ allows to add one time listeners with `addEventListener`
      ✓ supports the `removeEventListener` method
      ✓ wraps text data in a `MessageEvent`
      ✓ receives a `CloseEvent` when server closes (1000)
      ✓ receives a `CloseEvent` when server closes (4000)
      ✓ sets `target` and `type` on events
      ✓ passes binary data as a Node.js `Buffer` by default
      ✓ ignores `binaryType` for text messages
      ✓ allows to update `binaryType` on the fly
    SSL
      ✓ connects to secure websocket server
      ✓ connects to secure websocket server with client side certificate
      ✓ cannot connect to secure websocket server via ws://
      ✓ can send and receive text data
      ✓ can send a big binary message (56ms)
      ✓ allows to disable sending the SNI extension
      ✓ works around a double 'error' event bug in Node.js
    Request headers
      ✓ adds the authorization header if the url has userinfo
      ✓ honors the `auth` option
      ✓ favors the url userinfo over the `auth` option
      ✓ adds custom headers
      ✓ excludes default ports from host header
      ✓ doesn't add the origin header by default
      ✓ honors the `origin` option (1/2)
      ✓ honors the `origin` option (2/2)
    permessage-deflate
      ✓ is enabled by default
      ✓ can be disabled
      ✓ can send extension parameters
      ✓ can send and receive text data
      ✓ can send and receive a `TypedArray`
      ✓ can send and receive an `ArrayBuffer`
      ✓ consumes all received data when connection is closed abnormally
      ✓ handles a close frame received while compressing data
      #close
        ✓ can be used while data is being decompressed
      #send
        ✓ ignores the `compress` option if the extension is disabled
        ✓ calls the callback if the socket is closed prematurely
      #terminate
        ✓ can be used while data is being compressed
        ✓ can be used while data is being decompressed
    Connection close edge cases
      ✓ closes cleanly after simultaneous errors (1/2)
      ✓ closes cleanly after simultaneous errors (2/2)


  325 passing (2s)
  4 failing

  1) WebSocket
       Constants
         `CONNECTING`
           is enumerable property of class:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected

  {
+   configurable: true,
-   configurable: false,
    enumerable: true,
    value: 0,
+   writable: true
-   writable: false
  }
      + expected - actual

       {
      -  "configurable": true
      +  "configurable": false
         "enumerable": true
         "value": 0
      -  "writable": true
      +  "writable": false
       }
      
      at Context.<anonymous> (test/websocket.test.js:105:18)
      at processImmediate (node:internal/timers:464:21)

  2) WebSocket
       Constants
         `OPEN`
           is enumerable property of class:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected

  {
+   configurable: true,
-   configurable: false,
    enumerable: true,
    value: 1,
+   writable: true
-   writable: false
  }
      + expected - actual

       {
      -  "configurable": true
      +  "configurable": false
         "enumerable": true
         "value": 1
      -  "writable": true
      +  "writable": false
       }
      
      at Context.<anonymous> (test/websocket.test.js:105:18)
      at processImmediate (node:internal/timers:464:21)

  3) WebSocket
       Constants
         `CLOSING`
           is enumerable property of class:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected

  {
+   configurable: true,
-   configurable: false,
    enumerable: true,
    value: 2,
+   writable: true
-   writable: false
  }
      + expected - actual

       {
      -  "configurable": true
      +  "configurable": false
         "enumerable": true
         "value": 2
      -  "writable": true
      +  "writable": false
       }
      
      at Context.<anonymous> (test/websocket.test.js:105:18)
      at processImmediate (node:internal/timers:464:21)

  4) WebSocket
       Constants
         `CLOSED`
           is enumerable property of class:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected

  {
+   configurable: true,
-   configurable: false,
    enumerable: true,
    value: 3,
+   writable: true
-   writable: false
  }
      + expected - actual

       {
      -  "configurable": true
      +  "configurable": false
         "enumerable": true
         "value": 3
      -  "writable": true
      +  "writable": false
       }
      
      at Context.<anonymous> (test/websocket.test.js:105:18)
      at processImmediate (node:internal/timers:464:21)

```

Reply:
Ah I understand now, this happens due to https://github.com/websockets/ws/pull/1910#discussion_r663678860. I guess that is another reason to avoid that change.